Category Tree

ynzenu opened the pull request:
# Category Tree
Designed to help users find pathing categories.

I think it would be good to do a pre-release before doing a public release in order to monitor for any bugs/issues.

## General
- Greyed out text means the category is inactive and any child markers/trails are not rendered.
- Enabling/disabling categories in the context menu are synced to the category tree (UI Sync between the two).
- Most functionality is the same as in the context menu, such as when toggling "All Markers" the markers will be globally disabled via a setting.
- The "X hidden (click to show)" button is only displayed when "Indicate When Categories Are Hidden" is enabled in the settings.
<img src="https://github.com/user-attachments/assets/bc4cb646-8513-4e3c-a1ac-e36ff8a30bc0" width="600">

## Achievements
- Checkmark means the achievement is been completed, but only if "Allow Markers to Automatically Hide" is enabled in the settings.
- Category only displays a tooltip if "Show Tooltips for Achievements" is enabled in the settings.

<img src="https://github.com/user-attachments/assets/54df965c-10b0-4313-a2f2-f64944479cad" width="600">

## Path navigation
There are 3 ways to quickly navigate to a category in the tree:
- Right click on the category in the tree and select "Copy Path". This path can then be inserted in the search box.
<img src="https://github.com/user-attachments/assets/30a7a325-2aa7-4dff-926f-76da25637fa6" width="400">

- Right clicking on a marker on the map or mini map will display the "Open Parent Category" option. Clicking this will open the Category Tree and automatically navigate to the parent category of the selected marker.
<img src="https://github.com/user-attachments/assets/c0b5ca67-8641-4ae1-a045-0f6a306e3675" width="400">

- Right click on the category in the search results and select "Open In Category Tree".
<img src="https://github.com/user-attachments/assets/00ef1217-60d7-4a29-b8df-343142a76533" width="400">

## Icons
The entity textures are displayed side by side as can be seen in this example where the entities underneath the categories have 5 different textures:

<img src="https://github.com/user-attachments/assets/91f29c6a-bfe5-4e52-a2e7-077e492b384b" width="600">


## Missing
- Manifest has not been updated yet with a new version.
- "Truncate category name" setting is not used in the category tree)
- Localization - There is already localization missing in multiple places, so I thought this is something to add in another PR.

## Discussion Reference
https://discord.com/channels/531175899588984842/958231812780474409/1342005038624542731

## Related Github issues
[#99 [Suggestion] Quick action : show markers parent category in main menu ?](https://github.com/blish-hud/Pathing/issues/99)
[#90 Enhanced Marker UI](https://github.com/blish-hud/Pathing/issues/90)
[#43 Search for categories](https://github.com/blish-hud/Pathing/issues/43)

## Is this a breaking change?
No, I don't think so.

dlamkins commented on UI/Controls/NodeTree/TreeView.cs:
```diff
@@ -0,0 +1,304 @@
+                ReflowChildLayout(ChildBaseNodes);
+            } catch (Exception ex) {
+                //Investigate why collection is sometimes modified during reflow
+                _logger.Info($"Could not recalculate TreeView layout: {ex.Message}");
```
Why is this an INFO?

ynzenu replied:
I've now changed it to be a warning. 

It's not critical error, and it seems to be quite rare, which is why I haven't fully investigated it yet.

ynzenu commented:
I've made a few small tweaks and fixes:
- Fixed an issue that caused the 'X hidden (click to show)' number to be incorrect
- Fixed an issue that would create ghost categories when pasting paths
- Removed the 'Copy Display Name' option because it was a duplicate of 'Copy Name'

UI
- Completed achievements now show the checkmark on the right instead of left, and the checkbox is disabled. I found the icon in this PR: https://github.com/blish-hud/Pathing/pull/62
<img src="https://github.com/user-attachments/assets/df05a743-eac5-48a2-b1cb-5c543fedfbff" width="600">

- Added 2 labels for when the packs are not initialized (when you are in the character screen and haven't joined the game yet) and no packs are loaded (unchecked 'always load' or no packs installed).

<img src="https://github.com/user-attachments/assets/0acb45e7-c29d-4e09-82b8-08d12cc5f8a7" width="400">
<img src="https://github.com/user-attachments/assets/73f96fcd-6ac0-4604-9f02-f2916b2ed86e" width="400">
